### PR TITLE
feat: improve reboot dialog message

### PR DESCRIPTION
**Pull request by d-loose**
Since rebooting via systemd (see #206) doesn't respect any session inhibitors, we should remind the user to save any unsaved work before rebooting.

image

### CODE REVIEW

**Comment by tim-hm** [comment]
Can you ask @anasereijo for a servicing please? Given that clicking the button results in an immediate restart, I think we should double check with her and get some oversight on the copy as well ... eg is the immediacy sufficiently communicated in the copy + button color?

**Comment by anasereijo** [comment]
A couple of questions before suggesting copy:

- What happens if the user presses cancel? Is the update fully cancelled, and they have to start the process over again, or once they restart it will be done?

- Can we postpone the reboot and have a reminder? Something like 'remind me in -options: 10 minutes'

- If they restart without saving work, will they get a pop-up for each opened application with unsaved work?

**Comment by anasereijo** [comment]
one piece of feedback I can already give is about the buttons.

I'd keep both buttons grey. Even though we want the users to restart, in this situation I'd say that both buttons are the same level of importance. We don't want users to press 'restart' if they are not ready to (eg. if they haven't saved the files). Having restart as the primary button can lead some user to press on it without reading our advice first.

**Comment by anasereijo** [comment]
Thanks Dennis!

Just wondering if you read my first comment with some questions, so that I
can then propose copy?

Thank you

**Comment by d-loose** [comment]
> Just wondering if you read my first comment with some questions

Sorry, missed that one..


>  What happens if the user presses cancel? Is the update fully cancelled, and they have to start the process over again, or once they restart it will be done?

Only the restart will be cancelled, the update will be finished once they manually restart.


 
>  Can we postpone the reboot and have a reminder? Something like 'remind me in -options: 10 minutes'

Hm, that would probably need a long-lived process that runs in the background after the UI is closed.



>  If they restart without saving work, will they get a pop-up for each opened application with unsaved work?

Unfortunately not. This was the original idea using GNOME's own session manager, which provides this feature. However we can't access it through snapd yet, so we need to restart through a lower-level mechanism, which isn't aware of unsaved work. That's why Tim suggested improving the dialog message in the first place.



>  Is 'reboot' the term we consistently use? Or restart?

They're mostly being used interchangeably - do you think 'restart' is a better choice?

File: lib/src/l10n/app_en.arb [hunk 1]
```diff
@@ -94,7 +94,8 @@
   "ok": "OK",
   "olderVersions": "Older Versions",
   "reboot": "Reboot",
-  "rebootConfirm": "The update requires a reboot to complete. Do you want to reboot now?",
+  "rebootConfirmMessage": "The update requires a reboot to complete. Please save any unsaved work before proceeding. Do you want to reboot now?",
+  "rebootConfirmTitle": "Reboot required",
   "reinstall": "Reinstall",
```
Comment (tim-hm) [inline lib/src/l10n/app_en.arb:99]: GNOME shell uses the terminology 'Restart' so we should adopt that for consistency. If we need to revisit reboot vs restart (or update and upgrade) thats one for user research.

**Review by tim-hm — COMMENTED** [review]
No issues with the code but we need to revisit the copy and buttons.

- When you click Restart it is immediate and all open applications are force closed. I don't think this is clear enough in the text or buttons.
- The user can choose to "Restart later" meaning a perfectly valid option her is to dismiss and do it manually when they're ready.

How is this?

> This update requires a system restart. Make sure your device is plugged in or fully charged. Clicking "Restart Now" will force close all open applications and restart immediately. If you need more time, click "Restart Later".

Buttons: "Restart Now" or "Restart Later"

**Comment by d-loose** [comment]
Thanks, that's much clearer!

image

@anasereijo do you have any more feedback?

**Comment by anasereijo** [comment]
> Thanks, that's much clearer!
> 
> image
> 
> @anasereijo do you have any more feedback?

I do :) please bear with me

**Comment by anasereijo** [comment]
Thanks @tim-hm and @d-loose. 

It is much better; however, even though we are saying it will close all applications, we are not advising the users to save their work, and then we don't have pop-ups to do so when each application is closed for the restart. We want to make sure they are aware of the consequences of restarting now.

I propose the following:

**Restart to complete this update**

You must restart your device to complete the update. Make sure your device is plugged in or fully charged. Save your files before you restart to avoid losing data. 

[Restart later]  [Restart now]

**Review by tim-hm — APPROVED** [review]
LGTM when those copy changes ana suggested are made

**Comment by d-loose** [comment]
Final result

image

**Pull request merged**